Fix 20195 ux improvement for uuid values in search-input filters

DanielBiegler opened the pull request:
## Scope

What's changed:

- Previously we only updated the width when the value of the input matches our uuid regex because not all database vendors support the native uuid type like sqlite vs. postgresql. Now I preserved this behaviour but let the input width grow dynamically.
- Its clamped via magic numbers between `3ch - 40ch` width to not grow infinitely 


### Before:

[Screencast from 27.11.2023 16:35:34.webm](https://github.com/directus/directus/assets/14810858/86d5c06e-d215-4785-9242-fb97f7322a48)

### After:

[Screencast from 27.11.2023 16:32:56.webm](https://github.com/directus/directus/assets/14810858/528d384a-e16b-4cb9-8979-318cd5108f11)


## Potential Risks / Drawbacks

- Didnt change underlying behaviour, so only graphical issues like ugly growing, shrinking.

## Review Notes / Questions

- There can be the following issue:

[Screencast from 27.11.2023 16:24:40.webm](https://github.com/directus/directus/assets/14810858/ea240ef3-2e68-463f-87b2-8ca55ef94bce)

- This stems from two facts: We currently dont save partial (invalid) uuids to our bookmark/preset and because we currently have no way of uniquely identifying a specific filter inside the `preset.filter` the input gets lost upon rerendering. I believe this was discussed before as another separate UX improvement so this width issue should be resolved with that one.

---

Fixes partially #20195

DanielBiegler committed:
WIP input resizing

it works except for after dragging same input types above each other

DanielBiegler committed:
Merge branch 'main' into fix-20195-filter-width

DanielBiegler committed:
Merge remote-tracking branch 'origin/main' into fix-20195-filter-width

DanielBiegler committed:
make resizing work, replace method with native css

DanielBiegler committed:
Merge remote-tracking branch 'origin/main' into fix-20195-filter-width

rijkvanzanten commented on app/src/interfaces/_system/system-filter/input-component.vue:
```diff
@@ -39,9 +39,8 @@ const displayValue = computed(() => {
-const width = computed(() => {
-	return (props.value?.toString().length || 2) + 1 + 'ch';
-});
+const inputLength = ref<number>();
```
Seeing this input length is based on the HTML input value, which in turn is based on the `value` prop, we gotta make sure we reset this number when the value prop is updated from outside in with a watcher

DanielBiegler committed:
Create changeset

DanielBiegler committed:
add watcher for value change

DanielBiegler committed:
Merge remote-tracking branch 'refs/remotes/origin/fix-20195-filter-width' into fix-20195-filter-width

paescuj committed:
Format

paescuj committed:
Default / account for `null` for template ref

https://vuejs.org/guide/typescript/composition-api.html#typing-template-refs

paescuj committed:
Small clean-up

- Use newer defineEmits syntax
- Reordering
- Use watch with 'immediate' to get rid of fallback in 'width'
- Rename 'width' to 'inputWidth'

paescuj reviewed (approved):
The PR LGTM 🚀 

However, I think we should keep the linked issue open and come up with a solution to José's first point (in a separate PR 👍):
> show an error saying that input is not a valid UUID

Right now, there's no clear indication that a value is invalid, so a user can't really see why the filter is not applied.

A minimal solution might look like this:

<img width="400" src="https://github.com/directus/directus/assets/5363448/a7c2df01-d9e9-40b8-8dd0-d7a59c4d6a4a">

DanielBiegler commented:
Oh I like the idea with the border @paescuj ! Also thanks for including the template typing; TIL!
